src/generic_factory.h

```diff
@@ -125,7 +125,13 @@ class generic_factory
         // it's incremented when any changes to the inner id containers occur
         // version value corresponds to the string_id::_version,
         // so incrementing the version here effectively invalidates all cached string_id::_cid
-        int version = 0;
+        signed long long  version = 0;
+
+        void inc_version() {
+            do {
+                version++;
+            } while( version == -1 );
```

Comment:
Maybe the various instances of `-1` should be replaced by a named `static constexpr int64_t` defined in `string_id.h`?  Something like `INVALID_VERSION`?